contracts/FiatToken.cdc

```diff
@@ -1,1287 +1,258 @@
-                hashAlgorithm: HashAlgorithm.SHA3_256,
-                weight: pauserPubKeysWeights[i]
-            )
-            i = i + 1
-        }
-        pauserAccount.save(<- self.createNewPauser(publicKeys: pauserPubKeys, pubKeyAttrs: pubKeyAttrs), to: self.PauserStoragePath)
-        pauserAccount.link<&Pauser{OnChainMultiSig.PublicSigner}>(self.PauserPubSigner, target: self.PauserStoragePath)
-        pauserAccount.link<&Pauser{ResourceId}>(self.PauserUUIDPubPath, target: self.PauserStoragePath)
-        pauserAccount.link<&Pauser{PauseCapReceiver}>(self.PauserCapReceiverPubPath, target: self.PauserStoragePath)
-
-        // Assign the admin capabilities
-        let adminExecutorRef = contractAccount.borrow<&FiatToken.AdminExecutor>(from: self.AdminExecutorStoragePath)
-            ?? panic("cannot borrow AdminExecutor from storage")
-        let ownerExecutorRef = contractAccount.borrow<&FiatToken.OwnerExecutor>(from: self.OwnerExecutorStoragePath)
-            ?? panic("cannot borrow OwnerExecutor from storage")
-        adminExecutorRef.changeAdmin(to: adminAccount.address, newPath: initialAdminCapabilityPrivPath)
-        ownerExecutorRef.reassignOwner(to: ownerAccount.address, newPath: initialOwnerCapabilityPrivPath)
-        ownerExecutorRef.reassignMasterMinter(to: masterMinterAccount.address, newPath: initialMasterMinterCapabilityPrivPath)
-        ownerExecutorRef.reassignBlocklister(to: blocklisterAccount.address, newPath: initialBlocklisterCapabilityPrivPath)
-        ownerExecutorRef.reassignPauser(to: pauserAccount.address, newPath: initialPauserCapabilityPrivPath) 
-
+ 
         // Create a Vault with the initial totalSupply
         let vault <- create Vault(balance: self.totalSupply)
-        self.account.save(<-vault, to: self.VaultStoragePath)
+        self.account.storage.save(<-vault, to: self.VaultStoragePath)
+ 
+        // Create a Vault with the initial totalSupply
+        let minter <- create MinterResource()
+        self.account.storage.save(<-minter, to: self.MinterStoragePath)
 
         // Create public capabilities to the vault
-        contractAccount.link<&FiatToken.Vault{FungibleToken.Receiver}>(self.VaultReceiverPubPath, target: self.VaultStoragePath)
-        contractAccount.link<&FiatToken.Vault{FungibleToken.Balance}>(self.VaultBalancePubPath, target: self.VaultStoragePath)
-        contractAccount.link<&FiatToken.Vault{FiatToken.ResourceId}>(self.VaultUUIDPubPath, target: self.VaultStoragePath)
-
-        // Emit the TokensInitialized event
-        emit TokensInitialized(initialSupply: self.totalSupply)
-
+        let tokenCap = self.account.capabilities.storage.issue<&FiatToken.Vault>(self.VaultStoragePath)
+        self.account.capabilities.publish(tokenCap, at: self.VaultReceiverPubPath)
+        let receiverCap = self.account.capabilities.storage.issue<&FiatToken.Vault>(self.VaultStoragePath)
+        self.account.capabilities.publish(receiverCap, at: self.VaultBalancePubPath)
```

Comment:
With the [addition of entitlements and safe downcasting,](https://github.com/onflow/flips/blob/main/cadence/20221214-auth-remodel.md) anyone with a public reference to a resource, such as a `&{FungibleToken.Receiver}` reference to this resource, can cast that reference to the full concrete interface type, which exposes all the `access(all)` functions and fields in the resource. So in principle there is not problem with linking these as their concrete type `FiatToken.Vault`.

In order to restrict access to privileged functions like `withdraw()`, we use [entitlements](https://cadence-lang.org/docs/1.0/language/access-control). in order to call a privledged method like `withdraw()`, you need to be given a reference that is `auth(FungibleToken.Withdraw)`. So now there is no need to have different paths for different types because all the public functionality can be accessed from a single path.

I recommend that you read both of the documents that I linked so you can understand entitlements. Let me know if you have any questions!

Reply:
oh yes, I apologize. Good catch!